test/vfd_swmr_dsetops_writer.c

```diff
@@ -0,0 +1,1772 @@
+            "-w wwrites:      perform raw data modifications to all datasets\n"
+            "-u nticks:       `nticks` ticks for the reader to wait before verification\n"
+            "                 (default is 4)\n"
+            "-c csteps:       `csteps` steps communication interval between reader and writer\n"
+            "                 (default is 1)\n"
+            "-S:              do not use VFD SWMR\n"
+            "-N:              do not use named pipes for test synchronization\n"
+            "-q:              silence printouts, few messages\n"
+            "-b:              write data in big-endian byte order\n"
+            "                 (default is H5T_NATIVE_UINT32)\n\n"
+            "Note:\n"
+            "1. Require to specify at least -p, -g or -k option\n"
+            "2. -c <csteps> option cannot exceed -s <swrites> or -r <rwrites>\n"
+            "   or -l <lwrites> or -w <wwrites> option\n"
+            "\n",
+            progname);
+    exit(EXIT_FAILURE);
+} /* usage() */
+
+/*
+ * Initialize option info in state_t
+ */
+static bool
+state_init(state_t *s, int argc, char **argv)
+{
+    unsigned long tmp;
+    int           ch;
+    char          tfile[PATH_MAX];
+    char *        end;
+
+    *s = ALL_HID_INITIALIZER;
+    esnprintf(tfile, sizeof(tfile), "%s", argv[0]);
+    esnprintf(s->progname, sizeof(s->progname), "%s", basename(tfile));
+
+    while ((ch = getopt(argc, argv, "pte:gkm:n:s:r:l:w:bqSNu:c:")) != -1) {
+        switch (ch) {
+
+            case 'p': /* compact dataset */
+                s->compact = true;
+                break;
+
+            case 't': /* compact write */
+                s->compact_write = true;
+                break;
+
+            case 'g': /* contiguous dataset */
+                s->contig = true;
+                break;
+
+            case 'k': /* chunked datasets */
+                s->chunked = true;
+                break;
+
+            case 'q':
+                verbosity = 0;
+                break;
+
+            case 'b':
+                s->filetype = H5T_STD_U32BE;
+                break;
+
+            case 'S':
+                s->use_vfd_swmr = false;
+                break;
+
+            case 'N':
+                s->use_np = false;
+                break;
+
+            case 'e': /* # of elements for compact dataset */
+            case 'm': /* # of rows for -g and/or -k */
+            case 'n': /* # of cols for -g and/or -k */
+            case 's': /* # of sequential writes for -g and/or -k */
+            case 'r': /* # of random writes for -g and/or -k */
+            case 'l': /* # of hyperslab writes for -g and/or -k */
+            case 'w': /* # of raw data modifications for -g and/or -k */
+            case 'u': /* ticks for raeder to wait before verification */
+            case 'c': /* communication interval */
+                errno = 0;
+                tmp   = strtoul(optarg, &end, 0);
```

Comment:
Use HDF5 string operation like HDstrtoul? Also for the similar string operations?

Reply:
Done